### DISCUSSION TITLE
Solarforcast and last days energy consumption as calculation variables

### DISCUSSION THREAD

**Discussion by doncuco** [sonnen-tools/node-red#4] [Ideas]
The Idea is to be able to add the solar configuration and use the forecast as additional variable to the calculation or as condition to the decision flow.
In combination with the last few days average energy consumption (perhaps it might be necessary to differentiate between workdays and weekend days) and the battery capacity as additional  parameter the calculation oder decision flow would be more reliable for summer days.

**Comment by markusjnagel** [comment]
I like the idea, we could try it out... In the past, I have been playing with https://forecast.solar/ but the results were a bit of a mixed bag... they also keep changing over the day, when you request the same again and again.... yeah, I mean weather forecasts are not a precise science.... they are model-based forecasts...

We could give it a try and then we'll see what the results are.

With that said: Do you have an alternative that maybe provides better results? There are a couple of API providers around....

**Comment by doncuco** [comment]
I understand your concerns. And I have to admit that you are right. The quality of the solar forecast is very mixed. I have days where it perfectly matches. And days like today where it says I should get 6,6kwh and actually produced 3,5kwh. That is 50% of the prediction. 

As your calculation runs continiously it might be a solution to only use the forecast of the last hour in combination with the real production values of the last one to 3-5 hours. So you could at first check if the values for today are reliable. and then decide whether to use them.
You could also do that with the daily forecast. Check if the last days matched in a tolerable margin.

But no I do not have a better solution or never used an other forecaster. But I could do some research.

**Reply by doncuco** [reply]
In the End the actual idea is to use this data to reduce the target SOC for days where we have a solid forecast and know, that we won't empty our battery till we get free solar energy.
